src/OpenLoco/src/GameCommands/Terraform/LowerRaiseLandMountain.cpp

```diff
@@ -0,0 +1,405 @@
+        int8_t baseZDiff = cornerBaseZ - targetBaseZ;
+        uint8_t slopeFlags = 0;
+
+        if (baseZDiff > 0)
+        {
+            if (baseZDiff <= mtnToolHeightDiff)
+            {
+                return;
+            }
+
+            slopeFlags = lowerSurfaceCornerFlags(targetCorner, surface->slope());
+            targetBaseZ = surface->baseZ();
+
+            if (slopeFlags & SurfaceSlope::requiresHeightAdjustment)
+            {
+                targetBaseZ -= kSmallZStep;
+                slopeFlags &= ~SurfaceSlope::requiresHeightAdjustment;
+            }
+        }
+        else if (baseZDiff < 0)
+        {
+            if (-baseZDiff <= mtnToolHeightDiff)
+            {
+                return;
+            }
+
+            slopeFlags = raiseSurfaceCornerFlags(targetCorner, surface->slope());
+            targetBaseZ = surface->baseZ();
+
+            if (slopeFlags & SurfaceSlope::requiresHeightAdjustment)
+            {
+                targetBaseZ += kSmallZStep;
+                slopeFlags &= ~SurfaceSlope::requiresHeightAdjustment;
+            }
+        }
+        else // if (baseZDiff == 0)
+        {
+            return;
+        }
+
+        auto result = TileManager::adjustSurfaceHeight(pos, targetBaseZ, slopeFlags, removedBuildings, mtnToolGCFlags);
+        if (result != FAILURE)
+        {
+            mtnToolCost += result;
+        }
+    }
+
+    // 0x004633F6
+    static void adjustSurfaceSlopeSouth(Pos2 pos, int8_t targetBaseZ, std::set<Pos3, LessThanPos3>& removedBuildings)
+    {
+        adjustSurfaceSlope(pos, targetBaseZ, 2, SurfaceSlope::CornerUp::south, removedBuildings);
+    }
+
+    // 0x004634B9
+    static void adjustSurfaceSlopeWest(Pos2 pos, int8_t targetBaseZ, std::set<Pos3, LessThanPos3>& removedBuildings)
+    {
+        adjustSurfaceSlope(pos, targetBaseZ, 3, SurfaceSlope::CornerUp::west, removedBuildings);
+    }
+
+    // 0x0046357C
+    static void adjustSurfaceSlopeNorth(Pos2 pos, int8_t targetBaseZ, std::set<Pos3, LessThanPos3>& removedBuildings)
+    {
+        adjustSurfaceSlope(pos, targetBaseZ, 0, SurfaceSlope::CornerUp::north, removedBuildings);
+    }
+
+    // 0x0046363F
+    static void adjustSurfaceSlopeEast(Pos2 pos, int8_t targetBaseZ, std::set<Pos3, LessThanPos3>& removedBuildings)
+    {
+        adjustSurfaceSlope(pos, targetBaseZ, 1, SurfaceSlope::CornerUp::east, removedBuildings);
+    }
+
+    static uint32_t adjustMountainCentre(const LowerRaiseLandMountainArgs& args, std::set<Pos3, LessThanPos3>& removedBuildings, const uint8_t flags)
+    {
+        // Prepare parameters for raise/lower land tool
+        uint32_t result = FAILURE;
+        if (args.adjustment == 1)
+        {
+            RaiseLandArgs raiseArgs;
+            raiseArgs.centre = args.centre;
+            raiseArgs.pointA = args.pointA;
+            raiseArgs.pointB = args.pointB;
+            raiseArgs.corner = 4;
```

Comment:
do we have an enum for this? If not leave for later.

Reply:
No, not to my knowledge. The ones in SurfaceElement.h came to mind, but they seem to have another purpose, given `4` seems to be mean flat surface and not a corner.

Reply:
OpenRCT2 has this, which seems the appropriate enum. We don't seem to have an equivalent yet though:
```cpp
enum
{
    MAP_SELECT_TYPE_CORNER_0,
    MAP_SELECT_TYPE_CORNER_1,
    MAP_SELECT_TYPE_CORNER_2,
    MAP_SELECT_TYPE_CORNER_3,
    MAP_SELECT_TYPE_FULL,
    MAP_SELECT_TYPE_FULL_WATER,
    MAP_SELECT_TYPE_QUARTER_0,
    MAP_SELECT_TYPE_QUARTER_1,
    MAP_SELECT_TYPE_QUARTER_2,
    MAP_SELECT_TYPE_QUARTER_3,
    MAP_SELECT_TYPE_EDGE_0,
    MAP_SELECT_TYPE_EDGE_1,
    MAP_SELECT_TYPE_EDGE_2,
    MAP_SELECT_TYPE_EDGE_3,
};
```

Reply:
Fwiw, we do have this enum in Input.h. I think we should probably move to ViewportInteraction.h, though. The enum I quoted earlier could join it there.
```cpp
    enum class MapSelectionFlags : uint8_t
    {
        none = 0U,
        enable = 1U << 0,
        enableConstruct = 1U << 1,
        enableConstructionArrow = 1U << 2,
        unk_03 = 1U << 3,
        unk_04 = 1U << 4, // Vehicle orders?
        catchmentArea = 1U << 5,
        hoveringOverStation = 1U << 6,
    };
```

Reply:
Thats a different thing though the MapSelectionFlags are just activating certain overlays on the map.

Reply:
Yeah, I thought as much. Only meant to say we should introduce the other enum as well, and move both into ViewportInteraction.h